shields: ek_ra8d1_rtk7eka6m3b00001bu: add EK-RA8D1 to RTK7EKA6M3B00001BU adapter shield

thenguyenyf opened the pull request:
- Add the EK-RA8D1 to RTK7EKA6M3B00001BU Display Adapter Shield.
- Updated the EK-RA8D1 board documentation for the Parallel Display Expansion Usage section.

thenguyenyf committed:
shields: Add EK-RA8D1 to RTK7EKA6M3B00001BU Display Adapter

Add EK-RA8D1 to RTK7EKA6M3B00001BU Display Adapter.
This is a converter from EK-RA8D1 J57 port to RTK7EKA6M3B00001BU
Display CN1 pin signal.

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
boards: renesas: update ek_ra8d1 boards doc

Update usage guide when using the parallel graphics expandsion

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf commented:
Twister build failing due to https://github.com/zephyrproject-rtos/zephyr/issues/88396

thenguyenyf commented:
Hello @KhiemNguyenT , @thaoluonguw , @duynguyenxa . Could you please take a look?

soburi commented on include/zephyr/dt-bindings/gpio/renesas-ra-gpio-ioport.h:
```diff
@@ -15,4 +15,68 @@
 #define RENESAS_GPIO_DS_HIGH_SPEED_HIGH_DRIVE (0x2 << RENESAS_GPIO_DRIVE_POS)
 #define RENESAS_GPIO_DS_HIGH_DRIVE            (0x3 << RENESAS_GPIO_DRIVE_POS)
 
+/* EK-RA8D1 paralel graphics header */
```
This is no related with the SoC functionality.
So, please split it to separated file.

thenguyenyf replied:
I fixed it

soburi commented on include/zephyr/dt-bindings/gpio/renesas-ek-ra6m3g-header.h:
```diff
@@ -0,0 +1,37 @@
+/*
```
Instead of creating a header for each board, it is better to define the `gpio-map` in the dts for each board.
It would be a good idea to define the header accordingly as a connector pin definition.

Likes below.
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arduino/uno_r4/arduino_uno_r4_minima.overlay#L37-L64
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arduino/uno_r4/arduino_uno_r4_wifi.overlay#L37-L63

thenguyenyf replied:
I'm not sure that we can have 1 connector header define for both EK-RA8D1 and EK-RA6M3G. It's quite difference than UNO R3 connector because the pin order is difference on each board:
- EK-RA8D1: https://www.renesas.com/en/document/mat/ek-ra8d1-v1-user-manual?r=25452351 - 6.5 Parallel Graphics Expansion Port
- EK-RA6M3: https://www.renesas.com/en/document/mat/ek-ra6m3g-v1-users-manual?r=1168091 - 8.2 Signal Assignment

Although the EK-RA6M3G graphics LCD shield can be used on EK-RA8D1, you still need https://oshpark.com/shared_projects/pzfp0mCD to convert Parallel Graphics Connector of EK-RA8D1 to EK-RA6M3G graphics LCD signal.

soburi replied:
First of all, the purpose of the connector (nexus) is to provide definitions that can be used in the same way regardless of the board, so switching the headers with ifdef here would defeat that purpose.

You should put a connector definition with the pin arrangement set to suit each board in the DTS on the board side, because the board has this connector.

The name needs to be considered, but if a conversion board exists, you can also define the conversion board as a shield.
In this case, the converted connector name should be lcd_expansion_rtk7eka6m3b00001bu.

conversion board example:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/shields/rpi_pico_uno_flexypin/rpi_pico_uno_flexypin.overlay

thenguyenyf committed:
boards: renesas: ek_ra8d1: Add Parallel Graphics Expansion Connector

Add Parallel Graphics Connector node to EK-RA8D1 dts

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
samples: display: update rtk7eka6m3b00001bu shield related samples

The rtk7eka6m3b00001bu need addtional ek_ra8d1_rtk7eka6m3b00001bu shield
to build with ek_ra8d1/r7fa8d1bhecbd platform. This commit to update
extra args to select it when build with twister

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf commented on boards/renesas/ek_ra8d1/doc/index.rst:
```diff
@@ -102,8 +102,7 @@ Supported Features
 
-   - For using the Parallel Graphics Expansion Port (J57) with the Graphics Expansion Board supplied as part of the kit,
-     please set switch SW1 as following configuration:
+   - For using the Parallel Graphics Expansion Port (J57), please set switch SW1 as following configuration:
```
Done.

thenguyenyf commented on boards/shields/ek_ra8d1_rtk7eka6m3b00001bu/doc/img/top.webp:
Done.

thenguyenyf commented:
Hello @thaoluonguw , @duynguyenxa , @KhiemNguyenT . Could you please take a look?
